src/bci_build/package/__init__.py

```diff
@@ -402,12 +402,43 @@ def build_name(self) -> str | None:
 
     @property
     def build_version(self) -> str | None:
+        """Define the BuildVersion that is used for determining which container build
+        is newer than the other in the build service."""
+        # KIWI used to require an at least 3 component version X.Y.Z. For SLE, we set
+        # X.Y to MAJOR.MINOR and set Z to 0 for OsContainers. Derived
+        # containers inhert the base build_version X.Y and append a suffix .Z.ZZ
+
+        # It is important that the behavior for OsContainers is identical
+        # between KIWI and Dockerfile builds so that we can switch between these
+        # types.
+
+        if self.os_version.is_tumbleweed and isinstance(self, OsContainer):
+            return f"{str(datetime.datetime.now().year)}.0.0"
+        # TODO: also set it for non-kiwi type (historically we haven't done so)
+        if self.os_version.is_tumbleweed and self.build_recipe_type == BuildType.KIWI:
+            return f"{str(datetime.datetime.now().year)}.0"
+        if self.os_version.is_sle15 and isinstance(self, OsContainer):
```

Comment:
I don't quite get the `isinstance(self, OsContainer)` check here for the `.0` suffix

Reply:
see the comment above. osContainers have X.Y version numbers, derived containers have `X.Y.Z.ZZ` (`Z.ZZ` being derived from the containers tag_version (e.g. the application software version)